File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_cpp/DataStatistics2.cpp
#include <vector>
#include <cmath>
#include <algorithm>
#include <numeric>

class DataStatistics2 {
private:
    std::vector<double> data;

public:
    DataStatistics2(const std::vector<int>& input) {
        data.reserve(input.size());
        for (int val : input) {
            data.push_back(static_cast<double>(val));
        }
    }

    double get_sum() const {
        return std::accumulate(data.begin(), data.end(), 0.0);
    }

    double get_min() const {
        if (data.empty()) {
            return std::nan("");
        }
        return *std::min_element(data.begin(), data.end());
    }

    double get_max() const {
        if (data.empty()) {
            return std::nan("");
        }
        return *std::max_element(data.begin(), data.end());
    }

    double get_variance() const {
        if (data.empty()) {
            return std::nan("");
        }
        double mean = getSum() / data.size();
        double sum_sq = 0.0;
        for (double val : data) {
            double diff = val - mean;
            sum_sq += diff * diff;
        }
        return sum_sq / data.size();
    }

    double get_std_deviation() const {
        return std::sqrt(getVariance());
    }

    double get_correlation() const {
        return 1.0;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_py/code_output/DataStatistics2.py
import math
from typing import List


class DataStatistics2:
    def __init__(self, data: List[int]):
        self.data = [float(x) for x in data]

    def get_sum(self) -> float:
        return sum(self.data)

    def get_min(self) -> float:
        return min(self.data) if self.data else math.nan

    def get_max(self) -> float:
        return max(self.data) if self.data else math.nan

    def get_variance(self) -> float:
        if not self.data:
            return math.nan
        mean = self.get_sum() / len(self.data)
        return sum((val - mean) ** 2 for val in self.data) / len(self.data)

    def get_std_deviation(self) -> float:
        return math.sqrt(self.get_variance())

    def get_correlation(self) -> float:
        return 1.0